### PR TITLE
[Bugfix] fix java.lang.IllegalArgumentException: Self-suppression not permitted

### PR DESCRIPTION
**Pull request by 971758118**
fix java.lang.IllegalArgumentException: Self-suppression not permitted

Issues #5239

### CODE REVIEW

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/ProtoSerializer.java [hunk 1]
```diff
@@ -187,8 +187,12 @@ public void writeSerializedMessage(byte[] protoSerialized, String jsonSerialized
 
   @Override
   public void close() throws IOException {
-    output.flush();
-    idCache.clear();
+    try {
+      output.flush();
+      idCache.clear();
+    } catch (IOException e) {
+      throw new IOException(e);
```
Comment (jack-berg) [inline exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/ProtoSerializer.java:194]: I'm sorry for my ignorance - how exactly does this code result in `IllegalArgumentException` and how wrapping it in a try / catch solve it?

Comment (971758118) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/ProtoSerializer.java:194]: > I'm sorry for my ignorance - how exactly does this code result in `IllegalArgumentException` and how wrapping it in a try / catch solve it?

public abstract class Marshaler {
  ...
  public final void writeBinaryTo(OutputStream output) throws IOException {
    try (Serializer serializer = new ProtoSerializer(output)) {
      writeTo(serializer);
    }
  }

  protected abstract void writeTo(Serializer output) throws IOException;
  ...
}

in Marshaler.writeBinaryTo() , an exception may be thrown, and ProtoSerializer implements AutoCloseable, so ProtoSerializer.close() will be executed, but the same error is thrown in this method, same error instance, then Throwable.addSuppressed() will be executed, that's what caused this problem

    Throwable.java
    public final synchronized void addSuppressed(Throwable exception) {
        if (exception == this) {
            throw new IllegalArgumentException("Self-suppression not permitted", exception);
        } else if (exception == null) {
            throw new NullPointerException("Cannot suppress a null exception.");
        } else if (this.suppressedExceptions != null) {
            if (this.suppressedExceptions == SUPPRESSED_SENTINEL) {
                this.suppressedExceptions = new ArrayList(1);
            }
            this.suppressedExceptions.add(exception);
        }
    }

by try/catch with ProtoSerializer.close() and throw a new IOException instance might solve the problem
It worked in my test sample

**Comment by jack-berg** [comment]
Hmm... it looks like this [commit](https://github.com/971758118/opentelemetry-java/commit/d2a6e13d299470df27dc60310e035727a98ff483) is missing the user id required to validate CLA. Can you try to follow these [instructions](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) and then force push fresh commits?

If you can do this and the CLA is signed before tomorrow at about this time, I'll merge this PR to include it in this month's release.

**Comment by jack-berg** [comment]
Closing in favor of #5318.

**Pull request closed without merging**